cmd/mattermost/commands/team.go

```diff
@@ -247,3 +256,34 @@ func listTeamsCmdF(command *cobra.Command, args []string) error {
 
 	return nil
 }
+
+func searchTeamCmdF(command *cobra.Command, args []string) error {
+	a, err := InitDBCommandContextCobra(command)
+	if err != nil {
+		return err
+	}
+	defer a.Shutdown()
+
+	if len(args) < 1 {
+		return errors.New("Expected at least one argument. See help text for details.")
+	}
+
+	teams := getTeamsFromTeamArgs(a, args)
```

Comment:
Here you are getting teams based by name, this issue is for "search" teams. I think the better option here is to use directly the "app" layer. In this case, I think the option would be the SearchAllTeams function (in `app/team.go`), you can call it using `a.SearchAllTeams(searchTerm)`.

Reply:
I came up with this approach. Please do let me know your feedback over this before I commit the changes

    teams := make([]*model.Team, 0, len(args))

	for _, searchTerm := range args {
		team, err := a.SearchAllTeams(searchTerm)
		if err != nil{
			return err
		}
		teams = append(teams, team...)
	}

Reply:
Yes, that makes sense, probably after that you need to remove duplicated teams and sort them by name before printing.

Reply:
Thanks, I'm on it